internal/agent/idm_test.go

```diff
@@ -213,6 +214,19 @@ func (i *idM) serveAgent(w http.ResponseWriter, r *http.Request) {
 		i.error(w, http.StatusInternalServerError, "cannot read agent login request: %s", err)
 		return
 	}
+	ct, _, err := mime.ParseMediaType(r.Header.Get("Content-Type"))
+	if err != nil {
+		d := i.discharges[test]
+		d.c <- err
+		i.error(w, http.StatusBadRequest, "cannot parse mediatype: %s", err)
+		return
+	}
+	if ct != "application/json" {
+		d := i.discharges[test]
+		d.c <- err
```

Comment:
Perhaps factor a bunch of this code out into a separate function to avoid duplicating the four lines so many times?